### PR TITLE
Fix CI for forked repos

### LINKED ISSUE #805: CI does not run for forked repos

### PR DESCRIPTION
**Pull request by sindrig**
Simplify searching for good build and fix forks

closes #805

### CODE REVIEW

File: .github/actions/detection.ts [hunk 1]
```diff
@@ -35,23 +35,22 @@ export const findLastGoodBuild = async (
...
+      const previousValue = await previousValuePromise
+      if (previousValue) {
+        return previousValue
+      }
+      const successWorkflows = await workflowQueries.getData(currentValue)
+      let successOnBranch = getSuccessWorkflowsForBranch(successWorkflows)
+
+      return pickFirstMatchingSuccess(shas, successOnBranch)
+    }, null)) || {}
+  )
```
Comment (pshomov) [inline .github/actions/detection.ts:55]: Not quite sure this is more understandable for such a small set. ⚖️

**Review by pshomov — APPROVED** [review]
LGTM

File: .github/actions/detection.ts (hunk 1 above)
Comment (sindrig) [reply .github/actions/detection.ts]: What? This is javascipt, where all your crazy functional programming dreams can come true.

Simplified a bit.

**Review by pshomov — APPROVED** [review]
LGTM

**Review by pshomov — APPROVED** [review]

**Pull request merged**